### DISCUSSION TITLE
Suggestion: Clean and Semantic HTML for Remake of Admin Version

### DISCUSSION THREAD

**Discussion by ptamarshall** [solidusio/solidus#5161] [New Features or Ideas]
Hello Solidus.io team,

Firstly, I want to express my appreciation for the tremendous work you've done on the Solidus platform. Your commitment to creating a robust and flexible ecommerce platform is evident and commendable.

I'm writing to you today regarding the proposed remake of the admin version. While redesigns are exciting and can bring about significant improvement, there are some aspects I believe would greatly benefit the Solidus community if taken into consideration.

In my humble opinion, it would be best if the focus of the remake could be on clean, semantic HTML rather than catering to various theme preferences and incorporating CSS frameworks that, while sometimes helpful, can often complicate things.

Here's why I suggest this: every production-ready Solidus site inevitably needs its own stylesheets to cater to its unique brand identity and user experience. Therefore, if the new admin version could be built in a way that reuses whatever form CSS the user already has, it would simplify the process and avoid redundancy.

Furthermore, when considering reactive frontends, I believe that there currently is no better solution than StimulusReflex (in conjunction with Hotwired). This solution is incredibly powerful and efficient, and it would be fantastic if its usage wasn't complicated by the new admin version.

In conclusion, these suggestions aim at keeping the Solidus admin remake simple, efficient, and highly customizable for every user's unique needs. We all value the versatility and robustness of the Solidus platform, and I believe this approach would enhance these traits.

Thank you for considering my suggestions. I look forward to seeing how the Solidus admin remake unfolds.

Best regards,
P. Marshall

**Comment by waiting-for-dev** [comment]
Hi @ptamarshall,

We really appreciate your feedback. At this stage in our early development of the new admin, where some decisive decisions are being taken, the opinions of our users are very much needed :heart: 

> In my humble opinion, it would be best if the focus of the remake could be on clean, semantic HTML rather than catering to various theme preferences and incorporating CSS frameworks that, while sometimes helpful, can often complicate things.

We are committed to writing semantic HTML and focusing on accessibility, rowing in the same direction of a well-structured HTML.

When it comes to CSS frameworks, after much deliberation, we decided to go with TailwindCSS. We understand it comes with its strong points but also with some drawbacks. For us, we tipped towards using it for two important reasons:

- We're putting significant effort into making the new admin extensible; because of that, we are using view components. Having an interface divided into well-defined units with clear interfaces, which is pure Ruby, goes a long way toward it. TailwindCSS, with style defined within those same components, plays very well with it.
- TailwindCSS decreases by far the development time and effort. Solidus is an OS project with limited horsepower, so that's also very important to us.

> Therefore, if the new admin version could be built in a way that reuses whatever form CSS the user already has, it would simplify the process and avoid redundancy.

That's a really interesting idea. Of course, with all the CSS intricacies, it isn't easy to nest styles without other considerations. But we should investigate at least some way to do it with Tailwind (like injecting custom utilities). We would appreciate any idea here!!

> Furthermore, when considering reactive frontends, I believe that there currently is no better solution than StimulusReflex (in conjunction with Hotwired). This solution is incredibly powerful and efficient, and it would be fantastic if its usage wasn't complicated by the new admin version.

Stimulus and Hotwire will be part of the stack. Not sure we'll find the need for StimulusReflex, but we're open to use it if needed :slightly_smiling_face: 

I just want to say how helpful your feedback is at this point. Because of that, it would be very nice if you or anyone else could keep an eye on our [long-running branch for the new admin development](https://github.com/solidusio/solidus/pull/5068). We're creating all the PRs about the new admin against that branch. For instance, one covers [Tailwind extensibility](https://github.com/solidusio/solidus/pull/5080). You can also check the [README for the admin branch](https://github.com/solidusio/solidus/tree/nebulab/admin/admin) for the WIP version of the documentation.

Thanks @ptamarshall once again for your message!
